src/PublicShareSidebarTrigger.vue

```diff
@@ -0,0 +1,73 @@
+<!--
+  - @copyright Copyright (c) 2022 Daniel Calviño Sánchez <danxuliu@gmail.com>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<div class="button-holder">
+		<Button type="tertiary-on-primary"
+			:aria-label="ariaLabel"
+			@click="$emit('click')">
+			<template #icon>
+				<MenuPeople :size="20" />
+			</template>
+		</Button>
+	</div>
+</template>
+
+<script>
+import Button from '@nextcloud/vue/dist/Components/Button'
+import MenuPeople from './components/missingMaterialDesignIcons/MenuPeople.vue'
+
+export default {
+
+	name: 'PublicShareSidebarTrigger',
+
+	components: {
+		Button,
+		MenuPeople,
+	},
+
+	props: {
+		sidebarState: {
+			type: Object,
+			required: true,
+		},
+	},
+
+	computed: {
+		ariaLabel() {
+			if (this.sidebarState.isOpen) {
+				return t('spreed', 'Close Talk sidebar')
+			}
+
+			return t('spreed', 'Open Talk sidebar')
+		},
+	},
+
+}
+</script>
+
+<style scoped>
+.button-holder {
+	margin: 2px 5px 2px 2px;
```

Comment:
[Header contents are vertically centered](https://github.com/nextcloud/server/blob/87320cba99229e065747d699461b1dddc54326f8/core/css/header.scss#L196-L200). Are the vertical margins needed?

Out of curiosity, why 5px instead of 2px for `margin-right`?

Reply:
The margins where needed otherwise the box shadow was cut off.

For the right sidebar I added 3px on top so the white border of keyboard active doesn't touch the browser as it looked too weird.

Reply:
> The margins where needed otherwise the box shadow was cut off.

I tried removing the margins (or, rather, replacing with `margin-right: 3px`) and as far as I can tell it was working fine :shrug: That is why I was asking. But maybe it depends on the browser, I do not know.

> For the right sidebar I added 3px on top so the white border of keyboard active doesn't touch the browser as it looked too weird.

:+1: